From 86335ac679108bc85c27b442f59162e0d2633a76 Mon Sep 17 00:00:00 2001
From: Kristian Evensen <kristian.evensen@gmail.com>
Date: Thu, 22 Dec 2016 12:57:35 +0100
Subject: [PATCH] Set our default configuration options

We are disabling root login and require loging with keys in sshd_config.
We also need to avoid advertising public keys, public keys will instead
be specified explicitly.

In ssh_config, we need to time out ssh connection. We also need to set
ExitOnForwardFailure to make sure that ssh fails if ssh connection can be
established, but not tunnel.
---
 ssh_config  | 6 ++++++
 sshd_config | 6 +++---
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/ssh_config b/ssh_config
index 842ea866..d80e9d6d 100644
--- a/ssh_config
+++ b/ssh_config
@@ -17,6 +17,12 @@
 # list of available options, their meanings and defaults, please see the
 # ssh_config(5) man page.
 
+Host *
+    ServerAliveInterval 30
+    ServerAliveCountMax 3
+    ExitOnForwardFailure yes
+    PubkeyAuthentication no
+    IdentitiesOnly yes
 # Host *
 #   ForwardAgent no
 #   ForwardX11 no
diff --git a/sshd_config b/sshd_config
index 19b7c91a..6f22177a 100644
--- a/sshd_config
+++ b/sshd_config
@@ -29,12 +29,12 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin prohibit-password
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
 
-#PubkeyAuthentication yes
+PubkeyAuthentication yes
 
 # The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
 # but this is overridden so installations will only check .ssh/authorized_keys
@@ -54,7 +54,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #IgnoreRhosts yes
 
 # To disable tunneled clear text passwords, change to no here!
-#PasswordAuthentication yes
+PasswordAuthentication no
 #PermitEmptyPasswords no
 
 # Change to no to disable s/key passwords
-- 
2.25.1

